File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int num = n;
        int sum = 0;
        int prod = 1;

        while(num != 0){
            int ld = num%10;
            sum += ld;
            prod *= ld;
            num /=10;

        }
        int res = sum + prod;

        return ( n % res == 0) ? true : false;
    }
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int num = n;
        int sum = 0;
        int prod = 1;

        while(num != 0) {
            int dig = num % 10;

            sum += dig;
            prod *= dig;

            num /= 10;
        }

        int val = sum + prod;

        return (n % val == 0) ? true : false;
    }
}